packages/block-editor/src/components/block-inspector/index.js

```diff
@@ -246,6 +250,41 @@ const BlockInspectorSingleBlock = ( {
 		[ isSectionBlock, clientId ]
 	);
 
+	const {
+		selectedContentClientId,
+		isSelectedContentClientIdControlling,
+		isSelectedBlockInContentOnlyContainer,
+	} = useSelect( ( select ) => {
+		const {
+			getSelectedBlockClientId,
+			areInnerBlocksControlled,
+			getBlockRootClientId,
+			getTemplateLock,
+		} = select( blockEditorStore );
+
+		const _selectedBlockClientId = getSelectedBlockClientId();
+		const _isSelectedContentClientIdControlling = areInnerBlocksControlled(
+			_selectedBlockClientId
+		);
+
+		const rootClientId = getBlockRootClientId( _selectedBlockClientId );
+		const templateLock = getTemplateLock( rootClientId );
+
+		const _isSelectedBlockInContentOnlyContainer =
+			templateLock === 'contentOnly';
+
+		return {
+			selectedContentClientId: _selectedBlockClientId,
+			isSelectedContentClientIdControlling:
+				_isSelectedContentClientIdControlling,
+			isSelectedBlockInContentOnlyContainer:
+				_isSelectedBlockInContentOnlyContainer,
+			hasContentLockedParent: false,
+		};
+	} );
+	const { __unstableGetEditorMode } = useSelect( blockEditorStore );
+	const editorMode = __unstableGetEditorMode();
```

Comment:
We shouldn't call selectors during the render.

Here, we're not subscribing to store changes; we're only getting correct `editorMode` values because something else is triggering a rerender on store change. That's unreliable and can lead to bugs.